src/factories/artifact.ts

```diff
@@ -0,0 +1,77 @@
+import { BitmapText, Container } from 'pixi.js'
+import { barFactory } from '@/factories/bar'
+import { nodeLabelFactory } from '@/factories/label'
+import { Artifact } from '@/models/artifact'
+import { waitForConfig } from '@/objects/config'
+import { waitForCull } from '@/objects/culling'
+
+export type ArtifactFactory = Awaited<ReturnType<typeof artifactFactory>>
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export async function artifactFactory(artifact: Artifact) {
+  const element = new Container()
+  const content = new Container()
```

Comment:
The `content` Container holds the label and eventually will hold the icon as well. That way the combined dimensions of content can be used for sizing the bar.